### ISSUE TITLE
MD056 - Consistent column number check not taking into account escaped pipes

### ISSUE THREAD

**Issue by mbentley** [markdownlint/markdownlint#497]
## Description
An escaped pipe isn't taken into account when `MD056 - Table has inconsistent number of columns` is checking tables

### Environment
MacOS Sonoma 14.4.1

**MDL Version**
`0.13.0`

### Expected Behavior
Escaped pipes `\|` wouldn't be counted as a column.

### Actual Behavior
It appears to be seeing the pipe that is escaped and considering it a column.

## Replication Case
Here is a test file that shows the behavior

```md
# Test File

| First | Second |
| :---- | :----- |
| Normal | Normal |
| true\|false | test |
```

```sh
$ mdl test.md
test.md:6: MD056 Table has inconsistent number of columns
```

Rendered table:

| First | Second |
| :---- | :----- |
| Normal | Normal |
| true\|false | test |

**Comment by HariSekhon** [comment]
I'm hitting this bug too with the following:

```
| header  | header2 |
| --------|---------|
| normal  | normal  |
| `\|`    | Jump to column, without a prefix number jumps to start of line |
```

I tried working around it using triple backticks, but this doesn't work either:

```
| header  | header2 |
| --------|---------|
| normal  | normal  |
| ```\|``` | Jump to column, without a prefix number jumps to start of line |
```

We really need inline ignores to work around these false positives, like #16 or as a second choice the ability to exclude a whole file as per #167

**Comment by jaymzh** [comment]
This rule was already removed.
